stories/core/ExampleComponents/FreeDrawExample.js

```diff
@@ -12,37 +12,45 @@ export default class FreeDrawExample extends PureComponent {
     onDrawBegin: PropTypes.func,
     dataStyle: PropTypes.object,
     theme: PropTypes.object,
+    onDrawEnd: PropTypes.func,
+    allowMultipleShapes: PropTypes.bool,
   }
 
   static defaultProps = {
     theme: {},
     shapes: {},
+    allowMultipleShapes: false,
   }
 
-  constructor(props) {
-    super(props)
-    this.state = {
-      shapes: this.props.shapes,
-      enabled: false,
-    }
+  state = {
+    shapes: this.props.shapes,
+    enabled: this.props.allowMultipleShapes,
   }
 
   @autobind
   handleOnDrawEnd(shape) {
+    const { shapes } = this.state
+
+    if (this.props.onDrawEnd) this.props.onDrawEnd(shape)
+    const shapeId = Object.keys(shapes).length
     this.setState({
-      shapes: shape && { 0: shape },
-      enabled: false,
+      shapes: shape && { ...shapes, [shapeId]: shape },
```

Comment:
I thought you were removing this?

Reply:
I removed the logic from `handleClick` that updated the shapes in state. The way the component is written we have to use the state to keep all the shapes that have been drawn in the Multi-shape scenario. We can refactor this component to handle multi-shapes but since we're trying to get this released on Thursday I was trying to be as conservative as possible.